Neos.Media/Classes/Domain/Service/AssetVariantGenerator.php

```diff
@@ -93,16 +85,100 @@ public function createVariants(AssetInterface $asset): array
     }
 
     /**
-     * @param Image $originalAsset
+     * @param AssetInterface $asset
+     * @return AssetVariantInterface[] The created variants (if any), with the preset identifier as array key
+     * @throws AssetVariantGeneratorException
+     */
+    public function recreateVariants(AssetInterface $asset): array
+    {
+        // Currently only Image Variants are supported. Other asset classes can be supported, as soon as there is a common
+        // interface for creating and adding variants.
+        if (!$asset instanceof Image) {
+            return [];
+        }
+
+        $createdVariants = [];
+        foreach ($this->getVariantPresets() as $presetIdentifier => $preset) {
+            if ($preset->matchesMediaType($asset->getMediaType())) {
+                foreach ($preset->variants() as $variantIdentifier => $variantConfiguration) {
+                    $createdVariants[$presetIdentifier . ':' . $variantIdentifier] = $this->recreateVariant($asset, $presetIdentifier, $variantIdentifier);
+                }
+            }
+        }
+
+        return $createdVariants;
+    }
+
+    /**
+     * @param AssetInterface $asset
+     * @param string $presetIdentifier
+     * @param string $variantIdentifier
+     * @return AssetVariantInterface The created variant (if any)
+     * @throws AssetVariantGeneratorException
+     */
+    public function createVariant(AssetInterface $asset, string $presetIdentifier, string $variantIdentifier): ?AssetVariantInterface
+    {
+        // Currently only Image Variants are supported. Other asset classes can be supported, as soon as there is a common
```

Comment:
Wouldn't it make sense to change the signature to `createVariant(Image $image, ...)` and change it to the interface as soon as that exists? (And since `Image` will implement it, existing code should not break)

Reply:
Well, the concept is about *asset* variants, so… and again, here I followed the style this was implemented already. 🤷‍♂️